crates/storage/provider/src/bundle_state/state_reverts.rs

```diff
@@ -21,13 +21,27 @@ impl From<PlainStateReverts> for StateReverts {
 }
 
 impl StateReverts {
+    /// Write reverts to database, when [`StateReverts`] contains all values that will be inserted
+    /// for the given keys. See [`write_to_db_with_mode`](Self::write_to_db_with_mode).
+    pub fn write_to_db<TX: DbTxMut + DbTx>(
+        self,
+        tx: &TX,
+        first_block: BlockNumber,
+    ) -> Result<(), DatabaseError> {
+        self.write_to_db_with_mode(tx, first_block, true)
+    }
+
     /// Write reverts to database.
     ///
+    /// Passing `true` to to 'exhaustive_data_for_keys' parameter, will bring slight performance
+    /// gain, but will panic if values are not all that will be inserted for any given key.
+    ///
     /// Note:: Reverts will delete all wiped storage from plain state.
-    pub fn write_to_db<TX: DbTxMut + DbTx>(
+    pub fn write_to_db_with_mode<TX: DbTxMut + DbTx>(
         self,
         tx: &TX,
         first_block: BlockNumber,
+        exhaustive_data_for_keys: bool,
```

Comment:
can this be obtained with something similar to
```rust
let append_only = provider.count_entries::<tables::TransactionHashNumbers>()?.is_zero();
```
smthg like `entires, but "entries for a given key" on dupsort tables?

https://github.com/paradigmxyz/reth/blob/1c17f08ad209a1ae120436461ceb2f7f5391811c/crates/storage/db/src/abstraction/transaction.rs#L25-L26

Reply:
if you are talking about inlining that into the function, then i don't think the overhead for regular execution is worth it

Reply:
let's save this perf improvement for etl issue, my bad for bringing it up